pycbc/events/eventmgr.py

```diff
@@ -190,6 +190,23 @@ def __init__(self, opt, column, column_types, **kwds):
         self.template_events = numpy.array([], dtype=self.event_dtype)
         self.write_performance = False
 
+    def save_state(self, tnum_finished, filename):
+        """Save the current state of the background buffers"""
+        from six.moves import cPickle
+        self.tnum_finished = tnum_finished
+        logging.info('Writing checkpoint file at template {}'.format(tnum_finished))
+        cPickle.dump(self, open(filename, 'wb'), 
+                     protocol=cPickle.HIGHEST_PROTOCOL)
+
+    @staticmethod
+    def restore_state(filename):
+        """Restore state of the background buffers from a file"""
+        from six.moves import cPickle
+        mgr = cPickle.load(open(filename, 'rb'))
+        next_template = mgr.tnum_finished + 1
+        logging.info('Restarting with checkpoint at template {}'.format(next_template))
+        return mgr.tnum_finished + 1, mgr
+
```

Comment:
@ahnitz Not sure if this is relevant or not, but if you wanted to save the pickle to the output hdf file instead of a separate pickle file, you could use: https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/base_hdf.py#L828 to dump and https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/base_hdf.py#L887 to load. You'd have to move the functions out of `inference/io`, but there's no reason they couldn't live in `pycbc/io` instead.

Reply:
@cdcapano It might be, I'll think about it. If I can mix pickle / hdf, then there are some advantages, as I can handle explicitly some of the things that would be better stored in hdf as dataset, while keeping it otherwise agnostic to the structure of the class. 

In any case though, it would be a separate file. The format will look nothing like the final output file and won't be readable by codes that can read single triggers. Better to not make it look that might be possible.